helper/ldaputil/connection.go

```diff
@@ -0,0 +1,18 @@
+package ldaputil
+
+import (
+	"crypto/tls"
+
+	"github.com/go-ldap/ldap"
+)
+
+// Connection provides the functionality of an LDAP connection,
+// but through an interface.
+type Connection interface {
```

Comment:
This interface is new, intended to help with testing. While there is a test LDAP server out in the wild for everyone, it supports _reads only_, so it wouldn't allow for testing anything with a Modify method. It also differs slightly in behavior from Active Directory's LDAP in that you need more/different DN params to execute requests. So I think it'll overall increase test coverage if it's simple to test, and returning an interface is a simple way to do that.